### PR TITLE
Moves pending contributon into mojom definition

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/4620

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection

## Test Plan:
- enable rewards
- claim grant
- tip multiple unverified site
- make sure that you see pending box in your wallet
- open dialog
- try deleting single site
- try deleting all sites
- make sure that everything is carried out correctly

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_rewards/browser/rewards_service_impl.cc [hunk 1]
```diff
@@ -3034,32 +3034,30 @@ ledger::PendingContributionInfoList PendingContributionsOnFileTaskRunner(
 }
 
 PendingContributionInfo PendingContributionLedgerToRewards(
-    const ledger::PendingContributionInfo& contribution) {
+    ledger::PendingContributionInfoPtr contribution) {
```
Comment (tmancey) [inline components/brave_rewards/browser/rewards_service_impl.cc]: `contribution` should be `const`

File: components/brave_rewards/browser/rewards_service_impl.cc [hunk 2]
```diff
@@ -3034,32 +3034,30 @@ ledger::PendingContributionInfoList PendingContributionsOnFileTaskRunner(
...
+  info.added_date = contribution->added_date;
+  info.viewing_id = contribution->viewing_id;
+  info.expiration_date = contribution->expiration_date;
   return info;
 }
 
 void RewardsServiceImpl::OnGetPendingContributionsUI(
     GetPendingContributionsCallback callback,
-    const std::vector<std::string>& json_list) {
+    ledger::PendingContributionInfoList list) {
```
Comment (tmancey) [inline components/brave_rewards/browser/rewards_service_impl.cc]: `list` should be `const`

Comment (NejcZdovc) [reply components/brave_rewards/browser/rewards_service_impl.cc]: this one can't be const as then you get this error

```
../../brave/components/brave_rewards/browser/rewards_service_impl.cc:3060:44: error: call to deleted constructor of 'ledger::PendingContributionInfoPtr' (aka 'StructPtr<ledger::mojom::PendingContributionInfo>')
        PendingContributionLedgerToRewards(std::move(item));
                                           ^~~~~~~~~~~~~~~
../../mojo/public/cpp/bindings/struct_ptr.h:114:28: note: 'StructPtr' has been explicitly marked deleted here
  DISALLOW_COPY_AND_ASSIGN(StructPtr);
                           ^
../../brave/components/brave_rewards/browser/rewards_service_impl.cc:3037:46: note: passing argument to parameter 'contribution' here
    const ledger::PendingContributionInfoPtr contribution) {
                                             ^
1 error generated.
```

File: components/brave_rewards/browser/rewards_service_impl.cc (hunk 1 above)
Comment (NejcZdovc) [reply components/brave_rewards/browser/rewards_service_impl.cc]: done

File: components/brave_rewards/browser/rewards_service_impl.cc (hunk 2 above)
Comment (tmancey) [reply components/brave_rewards/browser/rewards_service_impl.cc]: Yeap apologies did not expand function

**Pull request merged**